client/models.py

```diff
@@ -0,0 +1,84 @@
+from django.db import models
+from django.core.validators import MinLengthValidator
+from django.core.exceptions import ValidationError
+
+
+class User(models.Model):
+    """An abstract class which holds basic user data:
+    Attributes:
+        user_name: a string which holds the user name, Also PK.
+        first_name: a string which holds the user first name.
+        last_name: a string which holds the user last name.
+        password: a string which holds the user password.
+        user_type: a string which holds the user type, either Supplier or Client.
+    """
+    user_name = models.CharField(max_length=32,
+                                 primary_key=True,
+                                 null=False,
+                                 validators=[MinLengthValidator(6)],
+                                 help_text="User Name")
+    first_name = models.CharField(max_length=32,
+                                  validators=[MinLengthValidator(1)],
+                                  help_text="First Name")
+    last_name = models.CharField(max_length=32,
+                                 validators=[MinLengthValidator(1)],
+                                 help_text="Last Name")
+    password = models.CharField(max_length=16,
+                                validators=[MinLengthValidator(6)],
+                                null=False,
+                                help_text="Password")
+    user_type = models.CharField(max_length=16,
```

Comment:
user is deleted,Please rebase your commit on upstream/main since User is on the buy_togehter_app import it from there and use it